modules/smaatoBidAdapter.js

```diff
@@ -61,95 +81,67 @@ const buildOpenRtbBidRequestPayload = (validBidRequests, bidderRequest) => {
-    at: 1,
-    imp,
-    cur: ['USD'],
-    tmax: bidderRequest.timeout,
-    site: {
-      id: window.location.hostname,
-      publisher: {
-        id: utils.deepAccess(validBidRequests[0], 'params.publisherId')
-      },
-      domain: window.location.hostname,
-      page: window.location.href,
-      ref: bidderRequest.refererInfo.referer
-    },
-    device: {
-      language: (navigator && navigator.language) ? navigator.language.split('-')[0] : '',
-      ua: navigator.userAgent,
-      dnt: utils.getDNT() ? 1 : 0,
-      h: screen.height,
-      w: screen.width
-    },
-    regs: {
-      coppa: config.getConfig('coppa') === true ? 1 : 0,
-      ext: {}
-    },
-    user: {
-      ext: {}
-    },
-    ext: {
-      client: CLIENT
+    if (bidderRequest.gdprConsent && bidderRequest.gdprConsent.gdprApplies === true) {
+      utils.deepSetValue(request, 'regs.ext.gdpr', bidderRequest.gdprConsent.gdprApplies ? 1 : 0);
+      utils.deepSetValue(request, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
     }
-  };
-
-  let ortb2 = config.getConfig('ortb2') || {};
 
-  Object.assign(request.user, ortb2.user);
-  Object.assign(request.site, ortb2.site);
+    if (bidderRequest.uspConsent !== undefined) {
+      utils.deepSetValue(request, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+    }
 
-  if (bidderRequest.gdprConsent && bidderRequest.gdprConsent.gdprApplies === true) {
-    utils.deepSetValue(request, 'regs.ext.gdpr', bidderRequest.gdprConsent.gdprApplies ? 1 : 0);
-    utils.deepSetValue(request, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
-  }
+    if (utils.deepAccess(validBidRequests[0], 'params.app')) {
+      const geo = utils.deepAccess(validBidRequests[0], 'params.app.geo');
+      utils.deepSetValue(request, 'device.geo', geo);
+      const ifa = utils.deepAccess(validBidRequests[0], 'params.app.ifa')
+      utils.deepSetValue(request, 'device.ifa', ifa);
+    }
 
-  if (bidderRequest.uspConsent !== undefined) {
-    utils.deepSetValue(request, 'regs.ext.us_privacy', bidderRequest.uspConsent);
-  }
+    const eids = utils.deepAccess(validBidRequests[0], 'userIdAsEids');
+    if (eids && eids.length) {
+      utils.deepSetValue(request, 'user.ext.eids', eids);
+    }
 
-  if (utils.deepAccess(validBidRequests[0], 'params.app')) {
-    const geo = utils.deepAccess(validBidRequests[0], 'params.app.geo');
-    utils.deepSetValue(request, 'device.geo', geo);
-    const ifa = utils.deepAccess(validBidRequests[0], 'params.app.ifa')
-    utils.deepSetValue(request, 'device.ifa', ifa);
+    return request
   }
 
-  const eids = utils.deepAccess(validBidRequests[0], 'userIdAsEids');
-  if (eids && eids.length) {
-    utils.deepSetValue(request, 'user.ext.eids', eids);
+  let payload = {};
+  if (validBidRequests.length === 1) {
+    payload = buildOpenRtbPayloadForIndividualBidRequest(validBidRequests[0])
+  } else {
+    payload = validBidRequests.map(buildOpenRtbPayloadForIndividualBidRequest)
```

Comment:
why can't we do just
`payload = validBidRequests.map(buildOpenRtbPayloadForIndividualBidRequest)`
for both cases? if it is one or multiple - shouldn't matter in this case